eoflow/models/classification_temp_nets.py

```diff
@@ -0,0 +1,169 @@
+import logging
+import tensorflow as tf
+from marshmallow import fields
+from marshmallow.validate import OneOf
+
+from .layers import ResidualBlock
+from .classification_base import BaseClassificationModel
+
+logging.basicConfig(level=logging.INFO,
+                    format='%(asctime)s %(levelname)s %(message)s')
+
+
+class TCNModel(BaseClassificationModel):
+    """ Implementation of the TCN network taken form the keras-TCN implementation
+
+        https://github.com/philipperemy/keras-tcn
+    """
+
+    class TCNModelSchema(BaseClassificationModel._Schema):
+        keep_prob = fields.Float(required=True, description='Keep probability used in dropout layers.', example=0.5)
+
+        kernel_size = fields.Int(missing=2, description='Size of the convolution kernels.')
+        nb_filters = fields.Int(missing=64, description='Number of convolutional filters.')
+        nb_stacks = fields.Int(missing=1)
+        dilations = fields.List(fields.Int, missing=[1, 2, 4, 8, 16, 32], description='Size of dilations used in the '
+                                                                                      'covolutional layers')
+        padding = fields.String(missing='CAUSAL', validate=OneOf(['CAUSAL', 'SAME']),
+                                description='Padding type used in convolutions.')
+        use_skip_connections = fields.Bool(missing=True, description='Flag to whether to use skip connections.')
+        return_sequences = fields.Bool(missing=False, description='Flag to whether return sequences or not.')
+        activation = fields.Str(missing='linear', description='Activation function used in final filters.')
+        kernel_initializer = fields.Str(missing='he_normal', description='method to initialise kernel parameters.')
+
+        use_batch_norm = fields.Bool(missing=False, description='Whether to use batch normalisation.')
+        use_layer_norm = fields.Bool(missing=False, description='Whether to use layer normalisation.')
+
+        class_weights = fields.List(fields.Float, missing=None, description='Class weights used in training.')
```

Comment:
This should be deleted cause it's defined in the base classfication schema.

Reply:
removed